Sources/SmithyTestUtil/CBORComparator.swift

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+import AwsCommonRuntimeKit
+import Foundation
+@_spi(SmithyReadWrite) import SmithyCBOR
+
+public struct CBORComparator {
+    /// Returns true if the logical CBOR values represented by the `Reader` instances are equal.
+    /// - Parameters:
+    ///   - dataA: The first CBOR data object to compare.
+    ///   - dataB: The second CBOR data object to compare.
+    /// - Returns: Returns true if the CBOR documents are equal.
+    public static func cborData(_ dataA: Data, isEqualTo dataB: Data) throws -> Bool {
+        let readerA = try Reader.from(data: dataA)
+        let readerB = try Reader.from(data: dataB)
+        return compareReaders(readerA, readerB)
+    }
```

Comment:
This type looks to me like it does the right things, but a few unit tests around it to verify basic cases around lists, maps, etc. would be useful.